File: DSA/C/Linear Search.c
/* -------------------------------------------------------------------------- */
/*                                Linear Search                               */
/* -------------------------------------------------------------------------- */

#include <stdio.h>
int main()
{
    int arr[10], i, n, element;
    printf("ENTER THE NO. OF ELEMENTS:");
    scanf("%d", &n);
    printf("ENTER ARRAY ELEMENTS:\n");
    for (i = 0; i < n; i++)
    {
        scanf("%d", &arr[i]);
    }
    printf("ENTERED ARRAY ELEMENTS ARE: \n");
    for (i = 0; i < n; i++)
    {
        printf("%d \n", arr[i]);
    }
    printf("ENTER THE ELEMENT TO BE SEARCHED:");
    scanf("%d", &element);
    for (i = 0; i < n; i++)
    {
        if (arr[i] == element)
        {
            printf("%d FOUND AT POSITION %d", element, i + 1);
            break;
        }
        if (i == n)
        {
            printf("%d NOT FOUND", element);
        }
    }
    return 0;
}

File: DSA/Java/LinearSearch.java
class LinearSearch{
    // method to find element from the array using linear search
    public static int linearSearching(int arr[], int target)
    {
        int n = arr.length;
        for (int i = 0; i < n; i++)
        {
           // check if element in the array is equal to the element which the user wants to search for 
            if (arr[i] == target)
                return i;
        }
        return -1;
    }
    //driver function
    public static void main(String args[])
    {
        int arr[] = {1,4,9,6,19,12,14,13,97};
        int target = 13;
 
        int ans = linearSearching(arr, target);
        if (ans == -1)
            System.out.print("\nElement is not present in array\n");
        else
            System.out.print("\nElement is present at index " + ans + "\n");
    }
}
